docs/samples/Microsoft.ML.Samples/Dynamic/GeneralizedAdditiveModels.cs

```diff
@@ -0,0 +1,128 @@
+            // feature value.
+
+            // Let's investigate the TeacherRatio variable. This is the ratio of students to teachers,
+            // so the higher it is, the more students a teacher has in their classroom.
+            // First, let's get the index of the variable we want to look at
+            var studentTeacherRatioIndex = featureNames.ToList().FindIndex(str => str.Equals("TeacherRatio"));
+
+            // Next, let's get the array of bin upper bounds from the model for this feature
+            var teacherRatioBinUpperBounds = gamModel.GetFeatureBinUpperBounds(studentTeacherRatioIndex);
+            // And the array of bin weights; these are the effect size for each bin
+            var teacherRatioFeatureWeights = gamModel.GetFeatureWeights(studentTeacherRatioIndex);
+
+            // Now, write the function to the console. The function is a set of bins, and the corresponding
+            // function values. You can think of GAMs as building a bar-chart lookup table.
+            //  Expected output:
+            //    Student-Teacher Ratio
+            //    x < 14.55 =>  2.105
+            //    x < 14.75 =>  2.326
+            //    x < 15.40 =>  0.903
+            //    x < 16.50 =>  0.651
+            //    x < 17.15 =>  0.587
+            //    x < 17.70 =>  0.624
+            //    x < 17.85 =>  0.684
+            //    x < 18.35 => -0.315
+            //    x < 18.55 => -0.542
+            //    x < 18.75 => -0.083
+            //    x < 19.40 => -0.442
+            //    x < 20.55 => -0.649
+            //    x < 21.05 => -1.579
+            //    x <   ∞   =>  0.318
+            //
+            // Let's consider this output. To score a given example, we look up the first bin where the inequality
+            // is satisfied for the feature value. We can look at the whole function to get a sense for how the
+            // model responds to the variable on a global level. For the student-teacher-ratio variable, we can see
+            // that smaller class sizes are predictive of a higher house value, while student-teacher ratios higher 
+            // than about 18 lead to lower predictions in house value. This makes intuitive sense, as smaller class 
+            // sizes are desirable and also indicative of better-funded schools, which could make buyers likely to
+            // pay more for the house.
+            //
+            // Another thing to notice is that these feature functions can be noisy. See student-teacher ratios > 21.05.
+            // Common practice is to use resampling methods to estimate a confidence interval at each bin. This will
+            // help to determine if the effect is real or just sampling noise. See for example 
+            // Tan, Caruana, Hooker, and Lou. "Distill-and-Compare: Auditing Black-Box Models Using Transparent Model 
+            // Distillation." <a href='https://arxiv.org/abs/1710.06169'>arXiv:1710.06169</a>."
+            Console.WriteLine(string.Empty);
```

Comment:
> Console.WriteLine(string.Empty); [](start = 11, length = 33)

This is technically correct, but weird way to move to new line.
You can just use Console.Writeline();
All across this file.